### DISCUSSION TITLE
Tuple and records in REPL

### DISCUSSION THREAD

**Discussion by jcubic** [babel/babel#14866] [Q&A] [answered]
I'm almost sure that there was an issue with the discussion about adding records and tuples proposals to REPL. Right now it's not working it gives an error:

> 'recordAndTuple' requires 'syntaxType' option whose value should be one of: 'hash', 'bar'

But I don't see such an option. Does REPL support those new data types?

**Comment by JLHwung** [comment, marked as answer]
This is https://github.com/babel/website/issues/1292

As a workaround, please use the `stage-2` preset in the left bar, then the REPL will use `hash` as the syntax type. In Babel 7.19 the `syntaxType` option will default to `hash` and we will remove the option in Babel 8.

You can also try out the [proposal playground](https://rickbutton.github.io/record-tuple-playground/#eyJjb250ZW50IjoiaW1wb3J0IHsgUmVjb3JkLCBUdXBsZSB9IGZyb20gXCJAYmxvb21iZXJnL3JlY29yZC10dXBsZS1wb2x5ZmlsbFwiO1xuY29uc3QgbG9nID0gY29uc29sZS5sb2c7XG5cbmNvbnN0IHJlY29yZCA9ICN7IHByb3A6IDEgfTtcbmNvbnN0IHR1cGxlID0gI1sxLCAyLCAzXTtcblxubG9nKFwiaXNSZWNvcmRcIiwgUmVjb3JkLmlzUmVjb3JkKHJlY29yZCkpO1xubG9nKFwiaXNSZWNvcmRcIiwgUmVjb3JkLmlzUmVjb3JkKHsgcHJvcDogMSB9KSk7XG5cbi8vIFNpbXBsZSBFcXVhbGl0eVxubG9nKFwic2ltcGxlXCIsXG4gICAgI3sgYTogMSB9ID09PSAjeyBhOjEgfSxcbiAgICAjWzFdID09PSAjWzFdKTtcblxuLy8gTmVzdGVkIEVxdWFsaXR5XG5sb2coXCJuZXN0ZWRcIiwgI3sgYTogI3sgYjogMTIzIH19ID09PSAjeyBhOiAjeyBiOiAxMjMgfX0pO1xuXG4vLyBPcmRlciBJbmRlcGVuZGVudFxubG9nKFwiIW9yZGVyXCIsICN7IGE6IDEsIGI6IDIgfSA9PT0gI3sgYjogMiwgYTogMX0pO1xuXG4vLyAtMCwgKzBcbmxvZyhcIi0wID09PSArMFwiLCAtMCA9PT0gKzApO1xubG9nKFwiI1stMF0gPT09ICNbKzBdXCIsICNbLTBdID09PSAjWyswXSk7XG5cbi8vIE5hTlxubG9nKFwiTmFOID09PSBOYU5cIiwgTmFOID09PSBOYU4pO1xubG9nKFwiI1tOYU5dID09PSAjW05hTl1cIiwgI1tOYU5dID09PSAjW05hTl0pO1xuIiwic3ludGF4IjoiaGFzaCIsImRvbU1vZGUiOmZhbHNlfQ==).

**Reply by jcubic** [reply]
I've used all stages and it was not working. Do I need to use only stage-2?

Anyway, I just use the second link for testing.

**Reply by JLHwung** [reply]
> I've used all stages and it was not working.

Indeed, it will be fixed in https://github.com/babel/babel/pull/14867.
